airflow/bin/cli.py

```diff
@@ -374,6 +375,9 @@ def run(args, dag=None):
             # the error when handlers do not have set_context defined.
             pass
 
+    sys.stdout = StreamLogWriter(log, logging.INFO)
```

Comment:
Perhaps it is better to use a context manager in case we returned or erred before having a chance to reset stdout/stderr? something like:

```
@contextmanager
def redirect_stdout():
    old_stdout = sys.stdout
    writer = StreamLogWriter(log, logging.INFO)
    try:
        yield writer
    finally:
        sys.stdout = old_stdout

@contextmanager
def redirect_stderr():
    // similar to above

with redirect_stdout(), redirect_stderr():
       // do stuff
    
```

Reply:
That can't (shouldn't) happen by design, as there is already a finally block. This structure was already there from 1.8.0-1.8.2

In addition the risk is pretty low as it happens in a forked process. I can move it into try/except block if needed, but as mentioned the code was here before.

Reply:
I don't see the finally block in this method but I do see one in the `helper` method. 

Overall I agree risk is low, and I'm fine with try/finally instead of a context manager, just some safety net :)

Reply:
Ah sorry you are looking at the cli method. I'll have a look